server/jobs/index.js

```diff
@@ -2,21 +2,50 @@
 /* eslint-disable global-require */
 import fs from 'fs';
 
+import { placementStatus, fetchNewPlacements } from '../modules/allocations';
+import client from '../helpers/redis';
+import { offBoardingResult, onBoardingResult } from './results';
+
 /**
  * @function getJobs
  * @param {string} folderName The folder containing the jobs.
  * @returns {array} An array of function representing jobs.
  */
 const getJobs = (folderName) => {
-  const jobs = fs
-    .readdirSync(`${__dirname}/${folderName}`)
-    .map((fileName) => {
-      const { default: exported } = require(`${__dirname}/${folderName}/${fileName}`);
-      return typeof (exported) === 'function' ? exported : () => {};
-    });
+  const jobs = fs.readdirSync(`${__dirname}/${folderName}`).map((fileName) => {
+    const { default: exported } = require(`${__dirname}/${folderName}/${fileName}`);
+    return typeof exported === 'function' ? exported : () => {};
+  });
   return jobs;
 };
 
-export const offboardingJobs = getJobs('offboarding');
+const offboardingJobs = getJobs('offboarding');
+
+const onboardingJobs = getJobs('onboarding');
 
-export const onboardingJobs = getJobs('onboarding');
+/**
+ * @desc Executes jobs to automate developer offboarding/onboarding tasks
+ *
+ * @param {string} type Type of job to execute: offboarding || onboarding
+ *
+ * @returns {void}
+ */
+export default async function executeJobs(type) {
+  const onboardingType = type === 'onboarding';
+  try {
+    const { rollingOff, onboarding } = placementStatus;
+    const category = onboardingType ? onboarding : rollingOff;
+    const newPlacements = await fetchNewPlacements(category, 7); // change 7 back to 1
+    return newPlacements.forEach(async (placement) => {
+      const jobList = onboardingType ? onboardingJobs : offboardingJobs;
+      // execute jobs in parallel but await all to complete before proceeding
+      await Promise.all(jobList.map(job => job(placement)));
+      client.incr('numberOfJobs');
+      console.log(offBoardingResult);
+      // commit offBoardingResult and onBoardingResult to database
+    });
+  } catch (error) {
+    // failed to fetch data from allocations: restart after 5 minutes
+    return setTimeout(() => executeJobs(type), 5 * 60000);
```

Comment:
Because every all code written before `fetchNewPlacements` are simple assignment statements, which can either have the actual value or `undefined`, whereas the program after `fetchNewPlacements` handles its own exceptions from within, which is the basis of `failure` or `success` report

Reply:
_...whereas the sub-program after `fetchNewPlacements` is assumed to handle its own exceptions..._

How about we don't assume???
Since the `fetchNewPlacements` is returning a promise, you can easily attach a `.then` or a `.catch` to it so you can be sure that you are handling *EXACTLY* what you need to.